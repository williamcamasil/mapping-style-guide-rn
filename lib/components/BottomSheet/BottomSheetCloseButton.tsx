import React from 'react';
import { TouchableOpacity } from 'react-native';

import { Images } from '../../assets';
import { useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';

type DialogCloseButtonPropsType = {
  onPress?: () => void;
  visible?: boolean;
  theme: AppThemeType;
};

const BottomSheetCloseButton: React.FC<DialogCloseButtonPropsType> = ({ onPress, visible, theme }) => {

  const buttonStyles = useViewStyles(() => [
    {
      position: 'absolute',
      top: theme.spacings.sSmall,
      right: theme.spacings.sSmall,
    },
  ], [theme.spacings.sSmall]);

  if (!visible) return null;

  return (
    <TouchableOpacity onPress={onPress} style={buttonStyles}>
      <Images.DialogClose width={36} height={36} />
    </TouchableOpacity>
  );
};

export default withTheme(BottomSheetCloseButton);
