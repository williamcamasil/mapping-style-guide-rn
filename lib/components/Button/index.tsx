import React, { ComponentType } from 'react';
import {
  StyleProp, StyleSheet, TouchableOpacity, TouchableWithoutFeedbackProps, View, ViewStyle,
} from 'react-native';

import { SvgProps } from 'react-native-svg';

import { useViewStyles } from '../../hooks/useStyles';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import Loading from '../Loading';
import Spacer from '../Spacer';
import Text, { TextPropTypes } from '../Text';
import {
  ButtonSizeType,
  ButtonVariantType,
  getSizeStyle,
  getTextColor,
  getVariantStyle,
  getStylesToBlock,
} from './utils';

export type ButtonPropsType<IconProps extends SvgProps = SvgProps> = {
  children?: TextPropTypes['children'];
  onPress: TouchableWithoutFeedbackProps['onPress'];
  style?: StyleProp<ViewStyle>;
  disabled?: TouchableWithoutFeedbackProps['disabled'];
  theme: AppThemeType;
  variant?: ButtonVariantType;
  size?: ButtonSizeType;
  loading?: boolean;
  testID?: string;
  Icon?: ComponentType<IconProps>;
  block?: boolean;
  containerStyle?: StyleProp<ViewStyle>;
  textColorVariant?: ButtonVariantType;
};

const styles = StyleSheet.create({
  containerIconAndText: {
    flexDirection: 'row',
    alignItems: 'center',
  },
});

const Button: React.FC<ButtonPropsType> = ({
  children,
  style,
  onPress,
  disabled,
  theme,
  variant,
  size,
  loading,
  testID,
  Icon,
  block,
  containerStyle,
  textColorVariant,
}) => {
  const hasText = Boolean(children);

  const buttonStyles = useViewStyles(
    () => [
      {
        display: 'flex',
        justifyContent: 'center',
        alignItems: 'center',
        borderRadius: theme.borders.radius.medium,
      },
      getVariantStyle(theme, variant, disabled),
      getSizeStyle(theme, size, hasText),
      style,
    ],
    [disabled, hasText, size, style, theme, variant],
  );

  const containerStyles = useViewStyles(() => [getStylesToBlock(block), containerStyle], [block, containerStyle]);

  const textColor = getTextColor(textColorVariant ? textColorVariant : variant, disabled);

  const renderText = () => {
    if (!hasText) return null;
    return (
      <Text
        testID="text-with-icon"
        variant={size === 'small' ? 'bodySmall' : 'body'}
        weight="bold"
        lineHeight="medium"
        color={textColor}
      >
        {children}
      </Text>
    );
  };

  const renderContent = () => {
    if (loading) {
      return <Loading color={textColor} />;
    }

    if (Icon) {
      return (
        <View testID="container-icon" style={styles.containerIconAndText}>
          <Icon width={32} height={32} color={theme.colors[textColor]} />
          {children ? <Spacer size={theme.spacings.sXXS} /> : null}
          {renderText()}
        </View>
      );
    }

    return renderText();
  };

  return (
    <View testID="container-button" style={containerStyles}>
      <TouchableOpacity
        testID={testID}
        style={buttonStyles}
        onPress={onPress}
        disabled={disabled || loading}
        activeOpacity={theme.opacities.intense}
      >
        {renderContent()}
      </TouchableOpacity>
    </View>
  );
};

Button.defaultProps = {
  disabled: false,
  variant: 'containedPrimary',
  size: 'medium',
  block: true,
};

export default withTheme(Button);
