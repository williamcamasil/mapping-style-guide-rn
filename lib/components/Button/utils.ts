import { ViewStyle } from 'react-native';

import { AppThemeType } from '../../theme/default';
import { ColorPaletesType } from '../../tokens';

export type ButtonVariantType =
  'containedPrimary' |
  'containedSecondary' |
  'outlinedPrimary' |
  'outlinedSecondary' |
  'text';

export type ButtonSizeType =
  'large' |
  'medium' |
  'small';

export const getVariantStyle = (
  theme: AppThemeType,
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ViewStyle => {
  switch (variant) {
    case 'containedSecondary':
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primary200,
        borderWidth: theme.borders.width.none,
      };
    case 'outlinedPrimary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.primary100 : theme.colors.primary600,
      };
    case 'outlinedSecondary':
      return {
        backgroundColor: 'transparent',
        borderWidth: theme.borders.width.thin,
        borderColor: disabled ? theme.colors.neutralGray100 : theme.colors.neutralGray300,
      };
    case 'text':
      return {
        backgroundColor: disabled ? theme.colors.neutralGray100 : 'transparent',
        borderWidth: theme.borders.width.none,
      };
    case 'containedPrimary':
    default:
      return {
        backgroundColor: disabled ? theme.colors.primary100 : theme.colors.primaryMain,
        borderWidth: theme.borders.width.none,
      };
  }
};

export const getTextColor = (
  variant?: ButtonVariantType,
  disabled?: boolean | null,
): ColorPaletesType => {
  switch (variant) {
    case 'containedSecondary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedPrimary':
      return disabled ? 'primary300' : 'primaryMain';
    case 'outlinedSecondary':
      return disabled ? 'neutralGray200' : 'neutralGray600';
    case 'text':
      return disabled ? 'neutralGray400' : 'neutralGray500';
    case 'containedPrimary':
    default:
      return disabled ? 'primary300' : 'neutralWhite';
  }
};

export const getSizeStyle = (theme: AppThemeType, size?: ButtonSizeType, hasText?: boolean): ViewStyle => {
  switch (size) {
    case 'large':
      return {
        height: 64,
        minWidth: 64,
        paddingHorizontal: hasText ? theme.spacings.sXL : undefined,
      };
    case 'small':
      return {
        height: 30,
        minWidth: 30,
        paddingHorizontal: hasText ? theme.spacings.sXS : undefined,
      };
    case 'medium':
    default:
      return {
        height: 48,
        minWidth: 48,
        paddingHorizontal: hasText ? theme.spacings.sMedium : undefined,
      };
  }
};

export const getStylesToBlock = (block: boolean | undefined): ViewStyle | null => {
  if (block) return null;

  return {
    flexDirection: 'row',
  };
};
