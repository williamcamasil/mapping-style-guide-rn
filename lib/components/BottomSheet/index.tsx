import React from 'react';
import {
  View,
  StyleSheet,
  StyleProp,
  ViewStyle,
  ScrollView,
} from 'react-native';

import { Edge, SafeAreaView } from 'react-native-safe-area-context';

import { useViewStyles } from '../../hooks';
import { withTheme } from '../../theme';
import { AppThemeType } from '../../theme/default';
import DialogActions from '../Dialog/DialogActions';
import DialogBase, { DialogBasePropsType } from '../Dialog/DialogBase';
import DialogExtraText from '../Dialog/DialogExtraText';
import DialogHeader from '../Dialog/DialogHeader';
import DialogMessage from '../Dialog/DialogMessage';
import DialogTitle from '../Dialog/DialogTitle';
import BottomSheetCloseButton from './BottomSheetCloseButton';

export type BottomSheetPropsType = DialogBasePropsType & {
  theme: AppThemeType;
  modalStyle?: StyleProp<ViewStyle>;
  containerStyle?: StyleProp<ViewStyle>;
  dismissible?: boolean;
};

const topSafeAreaViewEdges: Edge[] = ['top'];
const contentSafeAreaViewEdges: Edge[] = ['bottom'];

const styles = StyleSheet.create({
  container: {
    ...StyleSheet.absoluteFillObject,
    justifyContent: 'flex-end',
  },
  bottomSheet: {
    flexShrink: 1,
    width: '100%',
  },
});

const BottomSheet = ({
  theme,
  children,
  style,
  containerStyle,
  onDismiss,
  dismissible,
  ...others
}: BottomSheetPropsType): React.ReactElement => {
  const bottomSheetStyles = useViewStyles(() => [
    styles.bottomSheet,
    {
      borderTopLeftRadius: theme.borders.radius.XL,
      borderTopRightRadius: theme.borders.radius.XL,
      backgroundColor: theme.colors.neutralWhite,
    },
    style,
  ], [style, theme]);

  const scrollViewContentStyles = useViewStyles(() => [
    {
      padding: theme.spacings.sLarge,
    },
  ], [theme.spacings.sLarge]);

  const containerStyles = useViewStyles(() => [
    styles.container,
    containerStyle,
  ], [containerStyle]);

  return (
    <DialogBase
      {...others}
      onDismiss={onDismiss}
      dismissible={dismissible}
    >
      <View pointerEvents="box-none" style={containerStyles}>
        <SafeAreaView edges={topSafeAreaViewEdges}>
          {/*
          Impede que o bottom sheet entre atrás da status bar em conteúdos que geram rolagem,
          mas permite que entre atrás do botão "home" do iOS.
          */}
        </SafeAreaView>
        <View pointerEvents="box-none" style={bottomSheetStyles}>
          <SafeAreaView edges={contentSafeAreaViewEdges}>
            <View>
              <ScrollView
                contentContainerStyle={scrollViewContentStyles}
                bounces={false}
              >
                {children}
              </ScrollView>
              <BottomSheetCloseButton onPress={onDismiss} visible={dismissible} />
            </View>
          </SafeAreaView>
        </View>
      </View>
    </DialogBase>
  );
};

BottomSheet.Title = DialogTitle;
BottomSheet.Header = DialogHeader;
BottomSheet.Message = DialogMessage;
BottomSheet.ExtraText = DialogExtraText;
BottomSheet.Actions = DialogActions;

BottomSheet.defaultProps = {
  dismissible: true,
};

export default withTheme(BottomSheet);
