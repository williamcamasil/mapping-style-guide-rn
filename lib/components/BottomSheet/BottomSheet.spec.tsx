import 'react-native';
import React from 'react';

import { render } from '@testing-library/react-native';

import BottomSheet from '.';
import Button from '../Button';
import SampleHeaderSvg from '../Dialog/sample-header.svg';

describe('BottomSheet', () => {
  it('snapshot', () => {
    const tree = render(
      <BottomSheet
        isFocused
        visible
      >
        <BottomSheet.Header>
          <SampleHeaderSvg />
        </BottomSheet.Header>
        <BottomSheet.Title>
          Title
        </BottomSheet.Title>
        <BottomSheet.Message>
          Lorem ipsum is placeholder text commonly used in the graphic
        </BottomSheet.Message>
        <BottomSheet.Actions>
          <Button onPress={() => null}>
            Action 1
          </Button>
          <Button onPress={() => null}>
            Action 2
          </Button>
        </BottomSheet.Actions>
      </BottomSheet>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('snapshot invisible', () => {
    const tree = render(
      <BottomSheet
        isFocused={false}
        visible={false}
      >
        <BottomSheet.Title>
          Title
        </BottomSheet.Title>
        <BottomSheet.Message>
          Lorem ipsum is placeholder text commonly used in the graphic
        </BottomSheet.Message>
        <BottomSheet.Actions>
          <Button onPress={() => null}>
            Action 1
          </Button>
        </BottomSheet.Actions>
      </BottomSheet>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('snapshot dismissible false', () => {
    const tree = render(
      <BottomSheet
        isFocused
        visible
        dismissible={false}
      >
        <BottomSheet.Title>
          Title
        </BottomSheet.Title>
        <BottomSheet.Message>
          Lorem ipsum is placeholder text commonly used in the graphic
        </BottomSheet.Message>
        <BottomSheet.Actions>
          <Button onPress={() => null}>
            Action 1
          </Button>
        </BottomSheet.Actions>
      </BottomSheet>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
