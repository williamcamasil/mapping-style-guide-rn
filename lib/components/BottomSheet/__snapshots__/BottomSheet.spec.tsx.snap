// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BottomSheet snapshot 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "backgroundColor": "transparent",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
    testID="modal-safeArea"
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      style={
        {
          "backgroundColor": "#000000b3",
          "flex": 1,
          "opacity": 0,
        }
      }
      testID="modal-overlay"
    >
      <View
        collapsable={false}
        style={
          {
            "flex": 1,
            "marginTop": undefined,
            "transform": [
              {
                "translateY": 1334,
              },
            ],
          }
        }
        testID="modal-content"
      >
        <View
          accessibilityLabel="Fechar janela"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": false,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            {
              "bottom": 0,
              "justifyContent": "flex-end",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <RNCSafeAreaView
            edges={
              [
                "top",
              ]
            }
          />
          <View
            pointerEvents="box-none"
            style={
              {
                "backgroundColor": "#FFFFFF",
                "borderTopLeftRadius": 24,
                "borderTopRightRadius": 24,
                "flexShrink": 1,
                "width": "100%",
              }
            }
          >
            <RNCSafeAreaView
              edges={
                [
                  "bottom",
                ]
              }
            >
              <View>
                <RCTScrollView
                  bounces={false}
                  contentContainerStyle={
                    {
                      "padding": 28,
                    }
                  }
                >
                  <View>
                    <View
                      style={
                        {
                          "alignItems": "center",
                          "justifyContent": "center",
                          "marginBottom": 16,
                        }
                      }
                    >
                      <SvgMock />
                    </View>
                    <Text
                      allowFontScaling={false}
                      style={
                        {
                          "color": "#3E4149",
                          "flexShrink": 1,
                          "fontFamily": "Averta CY",
                          "fontSize": 23,
                          "fontWeight": "bold",
                          "letterSpacing": -0.02,
                          "lineHeight": 28,
                          "marginBottom": 12,
                          "textAlign": "left",
                        }
                      }
                    >
                      Title
                    </Text>
                    <Text
                      allowFontScaling={false}
                      style={
                        {
                          "color": "#676D79",
                          "flexShrink": 1,
                          "fontFamily": "Averta CY",
                          "fontSize": 16,
                          "fontWeight": "normal",
                          "letterSpacing": -0.02,
                          "lineHeight": 22,
                          "textAlign": "left",
                        }
                      }
                    >
                      Lorem ipsum is placeholder text commonly used in the graphic
                    </Text>
                    <View
                      style={
                        {
                          "marginTop": 40,
                        }
                      }
                    >
                      <View
                        style={{}}
                        testID="container-button"
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "alignItems": "center",
                              "backgroundColor": "#013188",
                              "borderRadius": 12,
                              "borderWidth": 0,
                              "display": "flex",
                              "height": 64,
                              "justifyContent": "center",
                              "marginTop": 0,
                              "minWidth": 64,
                              "opacity": 1,
                              "paddingHorizontal": 32,
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              {
                                "color": "#FFFFFF",
                                "flexShrink": 1,
                                "fontFamily": "Averta CY",
                                "fontSize": 16,
                                "fontWeight": "bold",
                                "letterSpacing": -0.02,
                                "lineHeight": 19,
                              }
                            }
                            testID="text-with-icon"
                          >
                            Action 1
                          </Text>
                        </View>
                      </View>
                      <View
                        style={{}}
                        testID="container-button"
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "alignItems": "center",
                              "backgroundColor": "transparent",
                              "borderRadius": 12,
                              "borderWidth": 0,
                              "display": "flex",
                              "height": 48,
                              "justifyContent": "center",
                              "marginTop": 16,
                              "minWidth": 48,
                              "opacity": 1,
                              "paddingHorizontal": 24,
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              {
                                "color": "#A4A7AF",
                                "flexShrink": 1,
                                "fontFamily": "Averta CY",
                                "fontSize": 16,
                                "fontWeight": "bold",
                                "letterSpacing": -0.02,
                                "lineHeight": 19,
                              }
                            }
                            testID="text-with-icon"
                          >
                            Action 2
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
                <View
                  accessibilityState={
                    {
                      "busy": undefined,
                      "checked": undefined,
                      "disabled": undefined,
                      "expanded": undefined,
                      "selected": undefined,
                    }
                  }
                  accessibilityValue={
                    {
                      "max": undefined,
                      "min": undefined,
                      "now": undefined,
                      "text": undefined,
                    }
                  }
                  accessible={true}
                  collapsable={false}
                  focusable={false}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    {
                      "opacity": 1,
                      "position": "absolute",
                      "right": 20,
                      "top": 20,
                    }
                  }
                >
                  <SvgMock
                    height={36}
                    width={36}
                  />
                </View>
              </View>
            </RNCSafeAreaView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`BottomSheet snapshot dismissible false 1`] = `
<View
  pointerEvents="box-none"
  style={
    {
      "backgroundColor": "transparent",
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    pointerEvents="box-none"
    style={
      {
        "flex": 1,
      }
    }
    testID="modal-safeArea"
  >
    <View
      accessibilityLiveRegion="polite"
      accessibilityViewIsModal={true}
      collapsable={false}
      style={
        {
          "backgroundColor": "#000000b3",
          "flex": 1,
          "opacity": 0,
        }
      }
      testID="modal-overlay"
    >
      <View
        collapsable={false}
        style={
          {
            "flex": 1,
            "marginTop": undefined,
            "transform": [
              {
                "translateY": 1334,
              },
            ],
          }
        }
        testID="modal-content"
      >
        <View
          accessibilityLabel="Fechar janela"
          accessibilityRole="button"
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": true,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessible={true}
          focusable={false}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "flex": 1,
            }
          }
        />
        <View
          pointerEvents="box-none"
          style={
            {
              "bottom": 0,
              "justifyContent": "flex-end",
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
        >
          <RNCSafeAreaView
            edges={
              [
                "top",
              ]
            }
          />
          <View
            pointerEvents="box-none"
            style={
              {
                "backgroundColor": "#FFFFFF",
                "borderTopLeftRadius": 24,
                "borderTopRightRadius": 24,
                "flexShrink": 1,
                "width": "100%",
              }
            }
          >
            <RNCSafeAreaView
              edges={
                [
                  "bottom",
                ]
              }
            >
              <View>
                <RCTScrollView
                  bounces={false}
                  contentContainerStyle={
                    {
                      "padding": 28,
                    }
                  }
                >
                  <View>
                    <Text
                      allowFontScaling={false}
                      style={
                        {
                          "color": "#3E4149",
                          "flexShrink": 1,
                          "fontFamily": "Averta CY",
                          "fontSize": 23,
                          "fontWeight": "bold",
                          "letterSpacing": -0.02,
                          "lineHeight": 28,
                          "marginBottom": 12,
                          "textAlign": "left",
                        }
                      }
                    >
                      Title
                    </Text>
                    <Text
                      allowFontScaling={false}
                      style={
                        {
                          "color": "#676D79",
                          "flexShrink": 1,
                          "fontFamily": "Averta CY",
                          "fontSize": 16,
                          "fontWeight": "normal",
                          "letterSpacing": -0.02,
                          "lineHeight": 22,
                          "textAlign": "left",
                        }
                      }
                    >
                      Lorem ipsum is placeholder text commonly used in the graphic
                    </Text>
                    <View
                      style={
                        {
                          "marginTop": 40,
                        }
                      }
                    >
                      <View
                        style={{}}
                        testID="container-button"
                      >
                        <View
                          accessibilityState={
                            {
                              "busy": undefined,
                              "checked": undefined,
                              "disabled": undefined,
                              "expanded": undefined,
                              "selected": undefined,
                            }
                          }
                          accessibilityValue={
                            {
                              "max": undefined,
                              "min": undefined,
                              "now": undefined,
                              "text": undefined,
                            }
                          }
                          accessible={true}
                          collapsable={false}
                          focusable={true}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            {
                              "alignItems": "center",
                              "backgroundColor": "#013188",
                              "borderRadius": 12,
                              "borderWidth": 0,
                              "display": "flex",
                              "height": 64,
                              "justifyContent": "center",
                              "marginTop": 0,
                              "minWidth": 64,
                              "opacity": 1,
                              "paddingHorizontal": 32,
                            }
                          }
                        >
                          <Text
                            allowFontScaling={false}
                            style={
                              {
                                "color": "#FFFFFF",
                                "flexShrink": 1,
                                "fontFamily": "Averta CY",
                                "fontSize": 16,
                                "fontWeight": "bold",
                                "letterSpacing": -0.02,
                                "lineHeight": 19,
                              }
                            }
                            testID="text-with-icon"
                          >
                            Action 1
                          </Text>
                        </View>
                      </View>
                    </View>
                  </View>
                </RCTScrollView>
              </View>
            </RNCSafeAreaView>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`BottomSheet snapshot invisible 1`] = `null`;
